doc/nistats_migration.rst

```diff
@@ -0,0 +1,84 @@
+.. _nistats_migration:
+
+===================================================
+A Quick Guide to Migrating Nistats Code to Nilearn
+===================================================
+
+A quick guide to changing your Nistats imports to work in Nilearn
+
+
+Datasets
+========
+Imports from
+`nistats.datasets <https://nistats.github.io/modules/reference.html#module-nistats.datasets>`_
+are in :mod:`nilearn.datasets`
+
+Hemodynamic Models
+==================
+Imports from
+`nilearn.stats.hemodynamic_models <https://nistats.github.io/modules/reference.html#module-nistats.hemodynamic_models>`_
+are in `nilearn.stats.first_level_model`
```

Comment:
removing the `:mod:` ( [as in last commit](https://github.com/nilearn/nilearn/pull/2359/commits/f92b54b0cf051b7876429e1e4a9c7731ae8e258d) ) is not the problem, you need to edit `reference.rst` to indicate this is a module so that the link target exists.

```diff
modified   doc/modules/reference.rst
@@ -418,8 +418,12 @@ uses.
     map_threshold
 
 
-nilearn.stats.first_level_model
--------------------------------
+:mod:`nilearn.stats.first_level_model`
+--------------------------------------
+
+.. automodule:: nilearn.stats.first_level_model
+   :no-members:
+   :no-inherited-members:
 
 
 **Classes**:
modified   doc/nistats_migration.rst
@@ -17,13 +17,13 @@ Hemodynamic Models
 ==================
 Imports from
 `nilearn.stats.hemodynamic_models <https://nistats.github.io/modules/reference.html#module-nistats.hemodynamic_models>`_
-are in `nilearn.stats.first_level_model`
+are in :mod:`nilearn.stats.first_level_model`
 
 Design matrix
 =============
 Most imports from
 `nistats.design_matrix <https://nistats.github.io/modules/reference.html#module-nistats.design_matrix>`_
-are in `nilearn.stats.first_level_model`
+are in :mod:`nilearn.stats.first_level_model`
 `nistats.design_matrix.make_second_level_design_matrix <https://nistats.github.io/modules/generated/nistats.design_matrix.make_second_level_design_matrix.html#nistats.design_matrix.make_second_level_design_matrix>`_
 is in :func:`nilearn.stats.second_level_model.make_second_level_design_matrix`
 
@@ -51,7 +51,7 @@ First Level Model
 =================
 Imports from
 `nistats.first_level_model <https://nistats.github.io/modules/reference.html#module-nistats.first_level_model>`_
-are in `nilearn.stats.first_level_model`
+are in :mod:`nilearn.stats.first_level_model`
 
 Second Level Model
 ==================

```

Reply:
Then sphinx fails when building `reference.rst`.

Reply:
That is the original source of this problem, that I havem't found a way to make Sphinx register `nilearn.stats.first_level_model` and `nilearn.stats.second_level_model` as (sub)modules.

Reply:
ok, thanks. It builds the documentation without failing locally so I opened a PR to see what happens in the CI. 
also, I pushed a commit upstream by mistake and reverted it, sorry

Reply:
@kchawla-pi  can you have a look at the built doc for #2413 ? the reference seems ok, see [here](https://5434-1235740-gh.circle-artifacts.com/0/doc/_build/html/modules/reference.html) . maybe since you had tried a new release of sphinx fixed the issue?